core/opendaq/reader/src/block_reader_impl.cpp

```diff
@@ -9,48 +9,72 @@ BEGIN_NAMESPACE_OPENDAQ
 using namespace std::chrono;
 using namespace std::chrono_literals;
 
-BlockReaderImpl::BlockReaderImpl(const SignalPtr& signal,
-                                 SizeT blockSize,
-                                 SampleType valueReadType,
-                                 SampleType domainReadType,
-                                 ReadMode mode)
+static SizeT calculateWrittenSamplesCount(SizeT initialWrittenSamplesCount, SizeT finalWrittenSamplesCount)
+{
+    auto writtenSamplesCount = finalWrittenSamplesCount > initialWrittenSamplesCount
+                                   ? finalWrittenSamplesCount - initialWrittenSamplesCount
+                                   : initialWrittenSamplesCount - finalWrittenSamplesCount;  // in case of counter wrapped
```

Comment:
What would be an example of that happening? Integer overflow?

Reply:
Removed this code discussion